pymeasure/instruments/mksinst/mks937b.py

```diff
@@ -0,0 +1,288 @@
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+#
+
+import re
+
+from pymeasure.instruments import Instrument
+from pymeasure.instruments.validators import strict_discrete_set
+from .adapters import MKSVISAAdapter
+
+
+_sensor_types = {"Cold Cathode": "CC",
+                 "Hot Cathode": "HC",
+                 "Pirani": "PR",
+                 "Convection Pirani": "CP",
+                 "Capacitance Manometer": "CM",
+                 "unknown (MB)": "MB",  # manual does not specify
+                 "unknown (GB)": "GB",  # manual does not specify
+                 "unknown (24)": "24",  # manual does not specify
+                 "no gauge": "NG",
+                 "no connection": "NC"}
+
+_ion_gauge_status = {"Wait": "W",
+                     "Off": "O",
+                     "Protect": "P",
+                     "Degas": "D",
+                     "Control": "C",
+                     "Rear panel Ctrl off": "R",
+                     "HC filament fault": "H",
+                     "No gauge": "N",
+                     "Good": "G",
+                     "NOT_IONGAUGE": "NAK152",
+                     }
+
+
+class MKS937B(Instrument):
+    """ MKS 937B vacuum gauge controller
+
+    :param adapter: pyvisa resource name of the instrument or an instance of
+                    MKSSerialAdapter, or MKSVISAAdapter.
+    :param address: device address included in every message to the instrument
+                    (default=253)
+    :param kwargs: Any valid key-word argument for Instrument
+    """
+
+    def __init__(self, adapter, address=253, **kwargs):
+        kwargs.setdefault("preprocess_reply", self.extract_reply)
+        if isinstance(adapter, str):
+            adapter = MKSVISAAdapter(adapter, **kwargs)
+        super().__init__(
+            adapter,
+            "MKS 937B vacuum gauge controller",
+            includeSCPI=False,
+            **kwargs
+        )
+        self.address = address
+        # compiled regular expression for finding numerical values in reply strings
+        self._re_fullresponse = re.compile(fr"@{self.address:03d}(.*)")
+        self._re_response_value = re.compile(fr"@{self.address:03d}ACK(.*)")
+        # set address in commands via dynamic properties
+        for prop in ["serial", "pressure1", "pressure2", "pressure3",
+                     "pressure4", "pressure5", "pressure6", "all_pressures",
+                     "combined_pressure1", "combined_pressure2",
+                     "sensor_typeA", "sensor_typeB", "sensor_typeC",
+                     "ion_gauge_status1", "ion_gauge_status3",
+                     "ion_gauge_status5", "unit",
+                     "power_status1", "power_status2", "power_status3",
+                     "power_status4", "power_status5", "power_status6",
+                     ]:
+            setattr(self, f"{prop}_command_process", self.command_process)
```

Comment:
You want to include the device address in the command sent to the device, don't you?
Alternatively, you can rewrite the instrument write/values, to do that for you. The values/read methods could check for those two additional "F" chars, as well.

That way you do not make all the properties dynamic.
```suggestion

    def values(self, command, **kwargs):
        ret = super().values(self.command_process(command), **kwargs)
        self._check_termination()
        return ret

    def _check_termination(self):
        """Check the read termination to correspond to the protocol."""
        t = self.adapter.read_bytes(2)  # read extra termination chars 'FF'
        if t != b'FF':
            raise ValueError(f"unexpected termination string received {t}")
```